internal/handler/handler.go

```diff
@@ -0,0 +1,349 @@
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error: with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployee(w http.ResponseWriter, r *http.Request) {
+	matches := employeeRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.Error(w, "error with response", http.StatusBadRequest)
+		return
+	}
+	e, err := h.service.GetEmployee(matches[1])
+	if err != nil {
+		if err.Error() == "get error: no this employee" {
+			http.Error(w, "error: not found", http.StatusNotFound)
+			return
+		}
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	jsonBytes, err := json.Marshal(e)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error: with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createPosition(w http.ResponseWriter, r *http.Request) {
+	var p internal.Position
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	if p.Salary == decimal.New(0, 0) || p.Name == "" {
+		http.Error(w, "error with request", http.StatusBadRequest)
+		return
+	}
+	err := h.service.CreatePosition(&p)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	jsonBytes, err := json.Marshal(p.ID)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error: with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (h *Handler) createEmployee(w http.ResponseWriter, r *http.Request) {
+	var e internal.Employee
+	if err := json.NewDecoder(r.Body).Decode(&e); err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	if e.LasName == "" || e.FirstName == "" {
+		http.Error(w, "error with request", http.StatusBadRequest)
+		return
+	}
+	err := h.service.CreateEmployee(&e)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	jsonBytes, err := json.Marshal(e.ID)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error: with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (h *Handler) updatePosition(w http.ResponseWriter, r *http.Request) {
+	var p internal.Position
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	err := h.service.UpdatePosition(&p)
+	if err != nil {
+		if err.Error() == "error incorrect input" {
```

Comment:
You should also track situation when you trying to update entity which is not exist and return 404 in that case

Reply:
+